server/src/main/java/io/spine/server/stand/UpdateHandler.java

```diff
@@ -25,100 +25,153 @@
+     * Obtains the handled subscription.
      */
-    private boolean includeAll() {
-        return target().getIncludeAll();
+    final Subscription subscription() {
+        return subscription;
     }
 
     /**
-     * Matches an event to the subscription filters.
+     * Tries to detect the passed event as a {@code SubscriptionUpdate} according
+     * to the {@code Subscription} criteria.
+     *
+     * @param event
+     *         the event to analyze against this subscription
+     * @return {@code SubscriptionUpdate} packed as {@code Optional} if the event matches the
+     *         subscription criteria, {@code Optional.empty()} otherwise
      */
-    private boolean matchByFilters(EventEnvelope event) {
-        return checkIdMatches(event) && checkEventMessageMatches(event);
-    }
+    abstract Optional<SubscriptionUpdate> detectUpdate(EventEnvelope event);
+
+    /**
+     * Extracts the ID value of the updated {@code Entity} or received {@code Event}.
+     *
+     * @param event
+     *         system event transmitting the {@code Entity} update info
+     *         or the domain event to which a subscription exists
+     */
+    abstract Any extractId(EventEnvelope event);
+
+    /**
+     * Tells whether the type of the updated {@code Entity} or received domain {@code Event}
+     * matches the type configured for the handled {@code Subscription}.
+     *
+     * @param event
+     *         system event transmitting the {@code Entity} update info
+     *         or the domain event to which a subscription exists
+     */
+    abstract boolean isTypeMatching(EventEnvelope event);
 
     /**
      * Checks if the event matches the subscription ID filter.
      */
-    private boolean checkIdMatches(EventEnvelope event) {
-        Any id = extractId(event);
+    boolean isIdMatching(EventEnvelope event) {
         TargetFilters filters = target().getFilters();
         IdFilter idFilter = filters.getIdFilter();
         boolean idFilterSet = !IdFilter.getDefaultInstance()
                                        .equals(idFilter);
         if (!idFilterSet) {
             return true;
         }
+        Any id = extractId(event);
         boolean result = idFilter.getIdList()
                                  .contains(id);
         return result;
     }
 
     /**
-     * Checks if the event message matches the subscription filters.
+     * Checks if the subscription has "include_all" clause.
      */
-    private boolean checkEventMessageMatches(EventEnvelope event) {
-        Message message = extractMessage(event);
-        TargetFilters filters = target().getFilters();
-        boolean result = filters
-                .getFilterList()
-                .stream()
-                .allMatch(f -> checkPasses(message, f));
-        return result;
+    boolean includeAll() {
+        return target().getIncludeAll();
+    }
+
+    /**
+     * "Activates" this callback with a given action.
```

Comment:
I'm concerned with the `Handler` name. Maybe callback would be better.

Reply:
There is a `callback` in this task already. Make sure you have reviewed the code.

Please suggest something more meaningful, if you can. I will think of it as well, but so far that's the best from simple names.